lib/web_ui/lib/src/engine/platform_dispatcher.dart

```diff
@@ -934,17 +935,17 @@ class EnginePlatformDispatcher extends ui.PlatformDispatcher {
   void _addFontSizeObserver() {
     const String styleAttribute = 'style';
 
-    _fontSizeObserver = createDomMutationObserver(allowInterop(
-        (List<dynamic> mutations, DomMutationObserver _) {
-      for (final dynamic mutation in mutations) {
-        final DomMutationRecord record = mutation as DomMutationRecord;
+    _fontSizeObserver = createDomMutationObserver(
+        (JSArray mutations, DomMutationObserver _) {
+      for (final DomMutationRecord mutation in mutations.toDart.cast<DomMutationRecord>()) {
```

Comment:
I think this odd cast here is needed because of a sub-optimal JS-interop layer; I've re-written this recently like so:

```
DomMutationObserver createDomMutationObserver(DomMutationCallbackFn fn) =>
    domCallConstructorString('MutationObserver', <Object>[
      allowInterop(
        (List<dynamic> mutations, DomMutationObserver observer) {
          fn(mutations.cast<DomMutationRecord>(), observer);
        },
      )
    ])! as DomMutationObserver;
```

(same trick as before, really)

Reply:
See above, I'm okay with this pattern, but it does mean an extra closure. Do you prefer the marginal amount of overhead  and slightly more ergonomics, vs this approach?

Reply:
I think we can leave this as is now, and eventually make it "nice" in package:web. Weak preference on this one.